force_gate_controller/include/force_gate_controller/tolerances.hpp

```diff
@@ -105,33 +105,7 @@ struct SegmentTolerances
  * \param params The ROS Parameters
  * \return Trajectory segment tolerances.
  */
-SegmentTolerances get_segment_tolerances(Params const & params)
-{
-  auto const & constraints = params.constraints;
-  auto const n_joints = params.joints.size();
-
-  SegmentTolerances tolerances;
-  tolerances.goal_time_tolerance = constraints.goal_time;
-
-  // State and goal state tolerances
-  tolerances.state_tolerance.resize(n_joints);
-  tolerances.goal_state_tolerance.resize(n_joints);
-  for (size_t i = 0; i < n_joints; ++i)
-  {
-    auto const joint = params.joints[i];
-    tolerances.state_tolerance[i].position = constraints.joints_map.at(joint).trajectory;
-    tolerances.goal_state_tolerance[i].position = constraints.joints_map.at(joint).goal;
-    tolerances.goal_state_tolerance[i].velocity = constraints.stopped_velocity_tolerance;
-
-    auto logger = rclcpp::get_logger("tolerance");
-    RCLCPP_DEBUG(
-      logger, "%s %f", (joint + ".trajectory").c_str(), tolerances.state_tolerance[i].position);
-    RCLCPP_DEBUG(
-      logger, "%s %f", (joint + ".goal").c_str(), tolerances.goal_state_tolerance[i].position);
-  }
-
-  return tolerances;
-}
+SegmentTolerances get_segment_tolerances(Params const & params);
```

Comment:
Is there a particular reason that you are moving these into a compilation unit. Just faster compilation?

Reply:
For now, we include "tolerances.hpp" and "velocity_jtc_parameters.hpp" from two places: the JTC and `force_gate_parent`. If we include an hpp file with definitions from multiple places, we get a "multiple definition" error (this is not prevented by the `#ifndef` guards).

Note that once we switch JTC to the inheritance model of the position/velocity controllers, these files will only be included from one place, and this change can be reverted.